### PR TITLE
Context-aware tour

### PR DESCRIPTION
**Pull request by microbit-matt-hillsdon**
Adds the ability to re-trigger a tour via a Help menu action,

To reduce the number of scenarios we require you to be connected to trigger the tour (prompted via dialog).

We show an adapted version of the "first data sample" tour steps in addition to the usual post-connect steps if you already have data samples when you connect.

Post-bulk-recording tours work again. Not sure they need text customisation but we could tweak the plural.

I've not tried to do a specialised tour for data import / "Open in micro:bit CreateAI" etc. so if you don't connect you do miss out.

### CODE REVIEW

File: src/components/HelpMenuItems.tsx [hunk 1]
```diff
@@ -102,4 +110,45 @@ const HelpMenuItems = () => {
   );
 };
 
+const tourMap = {
+  [createDataSamplesPageUrl()]: TourTrigger.Connect,
+  [createTestingModelPageUrl()]: TourTrigger.TrainModel,
+  // No UI to retrigger MakeCode tour
```
Comment (microbit-matt-hillsdon) [inline src/components/HelpMenuItems.tsx:116]: Perhaps, when re-triggering a tour via the button, we should clear the settings relating to subsequent tours.

File: src/pages/DataSamplesPage.tsx [hunk 2]
```diff
@@ -75,6 +75,7 @@ const DataSamplesPage = () => {
               {model ? (
                 <Button
                   onClick={handleNavigateToModel}
+                  className={tourElClassname.trainModelButton}
```
Comment (microbit-matt-hillsdon) [inline src/pages/DataSamplesPage.tsx:78]: We don't adapt the text in this case... but it seems hard to do well in terms of copy and important to talk about "Train model" nonetheless.

File: src/components/HelpMenuItems.tsx (hunk 1 above)
Comment (microbit-matt-hillsdon) [reply src/components/HelpMenuItems.tsx:116]: I still like this plan this morning.

File: src/pages/TestingModelPage.tsx [hunk 3]
```diff
@@ -45,6 +47,18 @@ const TestingModelPage = () => {
     }
   }, [model, navigateToDataSamples]);
 
+  const tourStart = useStore((s) => s.tourStart);
```
Comment (microbit-matt-hillsdon) [inline src/pages/TestingModelPage.tsx:50]: I'd like a better plan but I think that involves teaching the store about connections and routes.

File: src/store.ts [hunk 4]
```diff
@@ -808,25 +815,68 @@ const createMlStore = (logging: Logging) => {
...
-                  ? tourState
-                  : { id: TourId.DataSamplesPage, index: 0 },
-              }),
+              ({ actions, tourState, postConnectTourTrigger }) => {
+                return {
+                  actions:
+                    actions.length === 0 ? [createFirstAction()] : actions,
+
+                  // If a tour has been explicitly requested, do that.
+                  // Other tours are triggered by effects in the page so they run only on the correct screen.
```
Comment (microbit-matt-hillsdon) [inline src/store.ts:840]:
```suggestion
                  // Other tours are triggered by callbacks or effects on the relevant page so they run only on the correct screen.
```

File: lang/ui.en.json [hunk 5]
```diff
@@ -103,10 +103,6 @@
     "defaultMessage": "These are the MakeCode blocks for each detected action when you transfer your ML model and code to a micro:bit",
     "description": "Code column heading tooltip"
   },
-  "coming-soon": {
```
Comment (microbit-matt-hillsdon) [inline lang/ui.en.json:106]: Unused

**Comment by microbit-matt-hillsdon** [comment]
Review link: https://review-createai.microbit.org/retrigger-tour/

File: src/components/ActionDataSamplesCard.tsx [hunk 6]
```diff
@@ -155,18 +165,21 @@ const DataSamplesRowCard = ({
   );
 };
 
+interface RecordingAreaProps extends BoxProps {
```
Comment (microbit-grace) [inline src/components/ActionDataSamplesCard.tsx:168]: Do you mean `StackProps`?
```suggestion
interface RecordingAreaProps extends StackProps {
```

Comment (microbit-matt-hillsdon) [reply src/components/ActionDataSamplesCard.tsx:168]: I've not tended to do this when the client code doesn't need to / shouldn't be relying on a particular root-level component.

**Review by microbit-grace — APPROVED** [review]
LGTM

**Pull request merged**